app/hooks/useQuestionStateInUrl.ts

```diff
@@ -82,29 +90,31 @@ export default function useQuestionStateInUrl(minLogo: boolean, initialQuestions
       if (options?.moveToTop) {
         const removePageRe = new RegExp(`${pageid}.`, 'g')
         currentState = `${pageid}-${currentState.replace(removePageRe, '')}`
+        window.scrollTo({
+          top: 0,
+          behavior: 'smooth',
+        })
       }
 
-      const canonicallyAnsweredQuestions = canonicallyAnsweredQuestionsRef.current
-      if (canonicallyAnsweredQuestions.length === 0 && relatedQuestions.length > 0) {
-        // if canonicallyAnsweredQuestions (needed for relatedQuestions) are not loaded yet, wait a moment to re-run
+      const onSiteAnswers = onSiteAnswersRef.current
+      if (onSiteAnswers.length === 0 && relatedQuestions.length > 0) {
+        // if onSiteAnswers (needed for relatedQuestions) are not loaded yet, wait a moment to re-run
         setTimeout(() => toggleQuestion(questionProps, options), 500)
         return
       }
-      const canonicalQuestionTitleSet = new Set(
-        canonicallyAnsweredQuestions.map(({title}) => title)
-      )
+      const onSiteSet = new Set(onSiteAnswers.map(({pageid}) => pageid))
 
-      const newRelatedQuestions = relatedQuestions.filter((q) => {
-        const hasCanonicalAnswer = canonicalQuestionTitleSet.has(q.title)
-        // hide already displayed questions, detect duplicates by title (pageid can be different due to redirects)
+      const newRelatedQuestions = relatedQuestions.filter((question) => {
+        const isOnSite = onSiteSet.has(question.pageid)
+        // hide already displayed questions, detect duplicates by pageid (pageid can be different due to redirects)
         // TODO: #25 relocate already displayed to slide in as a new related one
-        const isAlreadyDisplayed = questions.some(({title}) => title === q.title)
-        return hasCanonicalAnswer && !isAlreadyDisplayed
+        const isAlreadyDisplayed = questions.some(({pageid}) => pageid === question.pageid)
+        return isOnSite && !isAlreadyDisplayed
       })
 
       const newState = getStateEntries(currentState)
         .map(([k, v]) => {
-          if (k === pageid) {
+          if (k === pageid.toString()) {
```

Comment:
isn't pageid always a string?

Reply:
should be after we re-generate tensorflow embeddings, but I figured if Coda doesn't guarantee the type of every value in a column, might be safer to keep this just in case future embeddigns will end up with accidental numbers too